gui.py

```diff
@@ -0,0 +1,189 @@
+"""
+This module contains implementation of graphical user interface, using PySide6
+
+"""
+from functools import lru_cache
+from typing import Generator
+import math
+
+from PySide6 import QtWidgets, QtGui, QtCore
+
+from model import GameMap, GameState
+from presenter import Presenter
+from config import ui as ucf
+from coordinates import Coordinates
+
+
+class Window(QtWidgets.QMainWindow):
+    """
+    Main window of user interface
+    """
+    def __init__(self, login_data, parent=None):
+        super().__init__(parent)
+        self.presenter_thread = Presenter(login_data)
+        self.presenter_thread.start()
+        self.screen = QtWidgets.QApplication.screenAt(self.pos())
+        self.setGeometry(0, 0, self.screen.size().width()//2, self.screen.size().height()//1.1)
+        self.setFixedSize(self.size())
+        self.hex_outer_radius = None
+        self.init_signals()
+
+    def get_mid_map(self) -> tuple[int, int]:
+        """
+        Give point to place central cell widget of map
+        :return: pixels x, y
+        """
+        mid_x = self.size().width() // 2
+        mid_y = self.size().height() // 2.2
+        return mid_x, mid_y
+
+    def init_signals(self) -> None:
+        """
+        Initiates signals from thread
+        :return: None
+        """
+        self.presenter_thread.game_state_updated.connect(self.refresh_map)
+        self.presenter_thread.game_ended.connect(self.show_message)
+
+    def set_hex_radius(self, map_size: int) -> None:
+        """
+        Set radius of Hex Widgets dependent on window size and map size
+        :param map_size: game map size (max coordinate value)
+        :return: None
+        """
+        self.hex_outer_radius =\
+            min(self.size().height() / (4 * map_size), self.size().height() / (4 * map_size))
+
+    def hex_to_pixel(self, cell: Coordinates) -> tuple[int, int]:
+        """
+        Calculates axis shift in pixels of each Hex Widget dependent on
+        coordinates relative to central Hex Widget
+        :param cell: Coordinates obj
+        :return: pixels shift (x, y)
+        """
+        shift_x = self.hex_outer_radius * (3 / 2 * cell[0])
+        shift_y = self.hex_outer_radius * (math.sqrt(3) / 2 * cell[0] + math.sqrt(3) * cell[1])
+        return shift_x, shift_y
+
+    def refresh_map(self, map_: GameMap, state: GameState) -> None:
```

Comment:
Why are you using trailing underscore here? If this is a way to avoid name clashing with some module level variable than you'd better think of other naming rather than use trailing underscore.